### PR TITLE
589 publish button

### PR DESCRIPTION
**Pull request by Axel-Jacobsen**
```
Publish button is working; however, I am getting a bug when I initially click on a new event from `localhost:8000/admin/events`. The first click doesn't redirect to the page, but raises a TypeError (`TypeError: undefined is not an object (evaluating 'props.entities.local[props.listItem.id].is_published'`) and changes the url to `localhost:8000/admin/events/<event id>`. Not sure of the solution, but will find one soon.
```
EDIT: This was solved in commit 157342f - Takes care of issue #589

### CODE REVIEW

**Comment by joncppl** [comment]
When it's a new item, `props.listItem.id` is null. It throws an error because null is being used as an array index.

I recommend checking whether it's null first. If it *is* null don't render the publish button (or render a dummy, disabled button).

File: dispatch/static/manager/src/js/constants/ActionTypes.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const TOPICS = resourceActionTypes('TOPICS')
 export const TAGS = resourceActionTypes('TAGS')
 export const TEMPLATES = resourceActionTypes('TEMPLATES')
 export const GALLERIES = resourceActionTypes('GALLERIES')
-export const EVENTS = resourceActionTypes('EVENTS', ['COUNT_PENDING'])
+export const EVENTS = resourceActionTypes('EVENTS', ['COUNT_PENDING', 'PUBLISH', 'UNPUBLISH'])
```
Comment (joncppl) [inline dispatch/static/manager/src/js/constants/ActionTypes.js]: `PUBLISH` and `UNPUBLISH` action types aren't used, so they aren't necessary here.

File: dispatch/static/manager/src/js/reducers/ToasterReducer.js [hunk 2]
```diff
@@ -184,6 +184,15 @@ export default function toasterReducer(toaster = {}, action) {
     return showToast(`${action.payload.length} event${action.payload.length > 1 ? 's' : ''} deleted`)
   case rejected(types.EVENTS.DELETE_MANY):
     return showToast('Some events could not be deleted', Intent.DANGER)
+  case fulfilled(types.EVENTS.PUBLISH):
```
Comment (joncppl) [inline dispatch/static/manager/src/js/reducers/ToasterReducer.js]: these publish actions are never fired off, and so these toast messages are never used.

**Review by joncppl — CHANGES_REQUESTED** [review]
Great work on your first front-end/react issue.

A few things in addition to 
the line comments.

1. publish button is unstyled
You can use the button component from the blueprint framework to make it look consistent
http://blueprintjs.com/docs/#core/components/button

2. Error when you try to publish a new, unsaved event.
Might as well hide it or disable it in this case.

**Comment by Axel-Jacobsen** [comment]
Did the suggestions from the review by @joncppl, and added a back button to audit page. In terms of the publish button styling, I wrote it as the "PRIMARY" style cause I thought blue would be good. I was also thinking that the success banner should be different? Same colour as publishing 
button, saying "Event Published" or "Event Unpublished". Thoughts?
screen shot 2017-06-29 at 2 49 27 pm

**Pull request merged**